tesseract_environment/src/utils.cpp

```diff
@@ -224,18 +179,21 @@ bool checkTrajectory(std::vector<tesseract_collision::ContactResultMap>& contact
 
   manager.applyContactManagerConfig(config.contact_manager_config);
 
+  tesseract_collision::ContactTrajectoryResults traj_contacts;
+  if (console_bridge::getLogLevel() > console_bridge::LogLevel::CONSOLE_BRIDGE_LOG_INFO)
+  {
+    traj_contacts = tesseract_collision::ContactTrajectoryResults(joint_names, static_cast<int>(traj.rows()));
+  }
+
   contacts.clear();
   contacts.reserve(static_cast<std::size_t>(traj.rows()));
 
-  /** @brief Making this thread_local does not help because it is not called enough during planning */
-  tesseract_collision::ContactResultMap state_results;
-  tesseract_collision::ContactResultMap sub_state_results;
```

Comment:
I would leave these here and use them as they were on dev. This has a significant performance impact on the number of times malloc is called.

Reply:
The problem was the clear method doesn't actually clear the map, just the vectors of collisions which were the values in the map. This caused false collisions to be recorded into the new trajectory logging struct which led to a segfault. Should I change the clear method to fully clear the map? The other option is to check the vector size in the debugging printout and call the `shrinkToFit()` method when necessary.

Reply:
I think the new logging needs to be updated to account for this. It is not okay to call shrinkToFit or update the clear method to clear the map because this defeats what this is intended to accomplishing. These change have a significant impact on both performance and number of malloc calls because the vector capacity remains the same even if clear is called preventing it from having to call malloc until the capacity is reached. If you clear the map or shrinkToFit the memory is released and it has to then request new memory. The contact result map class has methods for correctly checking the number of contacts.

Reply:
Added additional checks when printing the table at the end to make check for empty vectors.